File: Models/BoxData.cs
using System.Collections.Generic;

namespace BluePenguinMonitoring.Models
{
    public class BoxData
    {
        public List<ScanRecord> ScannedIds { get; set; } = new List<ScanRecord>();
        public int Adults { get; set; } = 0;
        public int Eggs { get; set; } = 0;
        public int Chicks { get; set; } = 0;
        public string? GateStatus { get; set; } = null; 
        public string Notes { get; set; } = "";
        public bool IsLocked { get; set; } = true;
    }
}

File: nestcheck_native/app/src/main/java/nz/co/penguinmonitor/model/BoxData.kt
package nz.co.penguinmonitor.model

import kotlinx.datetime.Clock
import kotlinx.datetime.Instant
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class BoxData(
    @SerialName("ScannedIds") val scannedIds: List<ScanRecord> = emptyList(),
    @SerialName("Adults") val adults: Int = 0,
    @SerialName("Eggs") val eggs: Int = 0,
    @SerialName("Chicks") val chicks: Int = 0,
    @SerialName("GateStatus") val gateStatus: String? = null,
    @SerialName("Notes") val notes: String = "",
    @SerialName("whenDataCollectedUtc") val whenDataCollectedUtc: Instant = Clock.System.now(),
    @SerialName("BreedingChance") val breedingChance: String? = null
)
